From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Axel Garcia K." <git@axelgk.com>
Date: Tue, 5 May 2026 18:57:43 +0000
Subject: [PATCH 1/4] Add SF2000 PicoDrive memory compatibility

Treat SF2000 like other no-mman libretro targets and make plat_mem_set_exec a no-op because UniFrog owns executable memory permissions for loaded core slots.
---
 platform/libretro/libretro-common/include/memmap.h | 2 +-
 platform/libretro/libretro.c                       | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/platform/libretro/libretro-common/include/memmap.h b/platform/libretro/libretro-common/include/memmap.h
index 5c03816054100d89a32de04409e20940dfc8cd05..e2f050702819830c0bb0e3f00cfd5eccd0a237f9 100644
--- a/platform/libretro/libretro-common/include/memmap.h
+++ b/platform/libretro/libretro-common/include/memmap.h
@@ -26,7 +26,7 @@
 #include <stdio.h>
 #include <stdint.h>
 
-#if defined(PSP) || defined(PS2) || defined(GEKKO) || defined(VITA) || defined(_XBOX) || defined(_3DS) || defined(WIIU) || defined(SWITCH) || defined(HAVE_LIBNX) || defined(__PS3__) || defined(__PSL1GHT__) || defined(__EMSCRIPTEN__)
+#if defined(PSP) || defined(PS2) || defined(GEKKO) || defined(VITA) || defined(_XBOX) || defined(_3DS) || defined(WIIU) || defined(SWITCH) || defined(HAVE_LIBNX) || defined(__PS3__) || defined(__PSL1GHT__) || defined(__EMSCRIPTEN__) || defined(SF2000)
 /* No mman available */
 #elif defined(_WIN32) && !defined(_XBOX)
 #include <windows.h>
diff --git a/platform/libretro/libretro.c b/platform/libretro/libretro.c
index 7612396ff0b7041c2928f2e6e2bf8b3bcf87f96c..e51c30d965f14d4aacdc7f2f5c3880f2712f0fd0 100644
--- a/platform/libretro/libretro.c
+++ b/platform/libretro/libretro.c
@@ -528,6 +528,10 @@ int plat_mem_set_exec(void *ptr, size_t size)
 
 #elif defined(VITA)
    ret = sceKernelOpenVMDomain();
+#elif defined(SF2000)
+   (void)ptr;
+   (void)size;
+   ret = 0;
 #else
    ret = mprotect(ptr, size, PROT_READ | PROT_WRITE | PROT_EXEC);
    if (ret != 0 && log_cb)

base-commit: f0d4a0118a9733a1f10bce5a4ac772c474f9300d
-- 
2.54.0

